src/Raven.Server/ServerWide/DeferrableTimeout.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using System.Threading.Tasks;
+using Sparrow.Threading;
+
+namespace Raven.Server.ServerWide;
+
+public class DeferrableTimeout
```

Comment:
Can we have some basic unit tests?

Reply:
Please check test for `ThrottledManualResetEventSlim`

Reply:
done